capstone/src/main/java/com/google/sps/servlets/student/StudentServlet.java

```diff
@@ -91,16 +88,31 @@ private static String convertToJsonUsingGson(StudentInfo info) {
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
     // TODO: All commented steps below
-    // 1. Add club to logged in student's club list if requested
-    // 2. Remove club from logged in student's club list if requested
-    // 3. Update student information with edited content
+    // 1. Remove club from logged in student's club list if requested
+    // 2. Update student information with edited content
 
     // No need to check if user's information is not in Datastore - this is done in doGet method
     // Get student object based on the logged in email
     UserService userService = UserServiceFactory.getUserService();
     String userEmail = userService.getCurrentUser().getEmail();
 
-    response.sendRedirect("/profile.html");
+    Query query = new Query(userEmail);
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+    Entity student = ImmutableList.copyOf(results.asIterable()).get(0);
+
+    String clubToJoin = request.getParameter(Constants.JOIN_CLUB_PROP);
+    if (clubToJoin != null && !clubToJoin.isEmpty()) {
+      // Add member to club's member list and update Datastore
+      Entity club = retrieveClub(clubToJoin, datastore);
```

Comment:
Will this always be a valid Entity? If not (eg: maybe it doesn't exist in the DataStore), then it would be good to check first

Reply:
I believe it will always be valid since we are only showing registered clubs on the Explore page, but I added a conditional for this and sent a "Bad Request" if the club does not exist just in case. Thank you for bringing this up!